src/main/java/org/varroa/chunk/command/generation/PublishPayloadGeneration.java

```
@@ -0,0 +1,109 @@
+        this.payloadGeneratorType = payloadGeneratorType;
+        this.message = message;
+    }
+
+    public static @NotNull PublishPayloadGeneration.Builder builder() {
+        return new PublishPayloadGeneration.Builder();
+    }
+
+    public int getPayloadGeneratorType() {
+        return payloadGeneratorType;
+    }
+
+    public @NotNull String getMessage() {
+        return message;
+    }
+
+    public @Nullable PayloadGenerator createGenerator(final int clientOffset, final @NotNull String clientId) {
+        // Some simple generators can be offset agnostic. With meaningful payloads, most aren't though.
+        if (payloadGeneratorType == GENERATOR_TYPE_RANDOM) {
+            return new RandomStringPayloadGenerator();
+        } else if (payloadGeneratorType == GENERATOR_TYPE_REGEX) {
+            return new RegexPayloadGenerator(clientOffset, message);
+        } else if ( payloadGeneratorType == GENERATOR_TYPE_MUSTACHE) {
+          return new MustachePayloadGenerator(clientId, message);
+        } else if (payloadGeneratorType == GENERATOR_TYPE_STATIC) {
+            return new StaticPayloadGenerator();
+        } else if (payloadGeneratorType == -1) {
+            log.error("No generator type specified in builder");
+        }
+        return null;
+    }
+
+    public static class Builder {
+
+        private int type = GENERATOR_TYPE_UNSPECIFIED;
+        private @Nullable String message;
+
+        private Builder() {
+        }
+
+        public @NotNull Builder type(final int type) {
+            this.type = type;
+            return this;
+        }
+
+        public @NotNull Builder message(final @NotNull String message) {
+            this.message = message;
+            return this;
+        }
+
+        public @NotNull PublishPayloadGeneration build() {
+            if(message == null) {
+                log.error("No payload message specified in payload generation builder");
```

please throw an exception